baremetal/metal3machine_manager.go

```diff
@@ -944,3 +972,19 @@ func (m *MachineManager) FindOwnerRef(refList []metav1.OwnerReference) (int, err
 	}
 	return 0, errors.New("OwnerRef not found")
 }
+
+func (m *MachineManager) updateObject(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+	err := m.client.Update(ctx, obj, opts...)
+	if apierrors.IsConflict(err) {
+		return &RequeueAfterError{}
+	}
+	return err
+}
+
+func (m *MachineManager) createObject(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+	err := m.client.Create(ctx, obj, opts...)
+	if apierrors.IsAlreadyExists(err) {
```

Comment:
createObject is called when the object does not exists, so when do you see it is possible while creating the object, the object might already exist?

Reply:
There is a small chance of race condition if the object gets created on the api server in between the get and create calls.

Reply:
Ok. There is almost no test codes also for the newly added functions or workflow. Is it difficult to produce such race conditions in tests?

Reply:
yes, I don't know how we could test it, without creating a flake. We would need to be able to pause the execution of the tested function after the get call and modify the object there. We could mock the client to return a 409. I'll think about it